### PR TITLE
Removing Sentiment Analysis & Fixing Images

### PR DESCRIPTION
**Pull request by kassidywebb**
Since images require a URL to post, they can't be automatically added to a message datastore. Blobstore takes care of this by turning the image URL into a blob (of type string) that we can store.

### CODE REVIEW

File: src/main/java/com/google/codeu/servlets/MessageServlet.java [hunk 1]
```diff
@@ -105,52 +93,58 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
        the link is then replaced with an img tag, a user is able to submit a
        pdf,png,jpg,gif,tiff, and bmp file */
     String text = Jsoup.clean(request.getParameter("text"), Whitelist.none());
-    String regex = "(https?://\\S+\\.(png|jpg|gif|pdf|tiff|bmp))";
+    String regex = "(https?://\\S+\\.(png|jpeg|gif|pdf|tiff|bmp))";
```
Comment (ryanluo) [inline src/main/java/com/google/codeu/servlets/MessageServlet.java]: let's add both jpg and jpeg

File: src/main/java/com/google/codeu/servlets/MessageServlet.java [hunk 2]
```diff
@@ -105,52 +93,58 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
     String privatemessage = request.getParameter("private");
-
-    float sentimentScore = this.getSentimentScore(textWithImagesReplaced);
-
-    /* This creates a Blobstore instance, then gets the image url(s) which are stored
-       in a map of string. Then converts the urls to a list. */
-    BlobstoreService blobstoreService = BlobstoreServiceFactory.getBlobstoreService();
-    Map<String, List<BlobKey>> blobs = blobstoreService.getUploads(request);
-    List<BlobKey> blobKeys = blobs.get("image");
+    System.out.println("Kassidy - " + request.getRequestURL());
```
Comment (ryanluo) [inline src/main/java/com/google/codeu/servlets/MessageServlet.java]: Let's remove these

File: src/main/java/com/google/codeu/servlets/MessageServlet.java [hunk 3]
```diff
@@ -105,52 +93,58 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
 
     /* Makes sure the list of images is not empty (and image was uploaded),
        then gets the url from Blobstore */
     if(blobKeys != null && !blobKeys.isEmpty()) {
-       BlobKey blobKey = blobKeys.get(0);
-       ImagesService imagesService = ImagesServiceFactory.getImagesService();
-       ServingUrlOptions options = ServingUrlOptions.Builder.withBlobKey(blobKey);
-       String imageUrl = imagesService.getServingUrl(options);
-       message.setImageUrl(imageUrl);
+      System.out.println("Kassidy is here");
```
Comment (ryanluo) [inline src/main/java/com/google/codeu/servlets/MessageServlet.java]: Let's delete this

File: src/main/java/com/google/codeu/servlets/MessageServlet.java [hunk 4]
```diff
@@ -105,52 +93,58 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
-    else {
+    } else {
       recipient = recipient;
     }
 
-    //add empty string as imageUrl parameter
-    Message message = new Message(user, textWithImagesReplaced, recipient, sentimentScore, "");
+    Message message = new Message(user, textWithImagesReplaced, recipient);
+
+    /* This creates a Blobstore instance, then gets the image url(s) which are stored
```
Comment (ryanluo) [inline src/main/java/com/google/codeu/servlets/MessageServlet.java]: Consider putting all the blobstore writes into a separate function?

File: src/main/java/com/google/codeu/servlets/MessageServlet.java [hunk 5]
```diff
@@ -105,52 +93,58 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
-       message.setImageUrl(imageUrl);
+      System.out.println("Kassidy is here");
+      BlobKey blobKey = blobKeys.get(0);
+      ImagesService imagesService = ImagesServiceFactory.getImagesService();
+      ServingUrlOptions options = ServingUrlOptions.Builder.withBlobKey(blobKey);
+      try {
+         String imageUrl = imagesService.getServingUrl(options);
+         message.setImageUrl(imageUrl);
+      } catch (ImagesServiceFailureException unused) {
+
```
Comment (ryanluo) [inline src/main/java/com/google/codeu/servlets/MessageServlet.java]: Log the exception here.

**Review by ryanluo — CHANGES_REQUESTED** [review]
Nice work! Just some small changes.

File: src/main/java/com/google/codeu/servlets/MessageServlet.java (hunk 4 above)
Comment (kassidywebb) [reply src/main/java/com/google/codeu/servlets/MessageServlet.java]: Fixed.

File: src/main/java/com/google/codeu/servlets/MessageServlet.java [hunk 6]
```diff
@@ -117,7 +117,16 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
 
-    /* This creates a Blobstore instance, then gets the image url(s) which are stored
+    //blobstore function that gets and saves image url
+    setMessageImageUrl(request, message);
+    datastore.storeMessage(message);
+
+    response.sendRedirect("/user-page.html?user=" + user);
+  }
+
+  private void setMessageImageUrl (HttpServletRequest request, Message message) {
```
Comment (Ajide49) [inline src/main/java/com/google/codeu/servlets/MessageServlet.java:129]: I would add a quick function description of what's happening in setMessageImageUrl

**Comment by Ajide49** [comment]
Everything seems good to me! I would just add a comment header explaining setImageUrl in the MessageServlet.java, good job!

**Comment by imanityson** [comment]
Everything seems cool, maybe just something explaining how image plays into messaging?

**Review by ryanluo — APPROVED** [review]

**Pull request merged**